pkg/pillar/devicenetwork/dns.go

```diff
@@ -166,12 +166,17 @@ func ResolveWithPortsLambda(domain string,
 
 	wgChan := make(chan struct{})
 
+	wgWaitDone := make(chan struct{})
```

Comment:
Is it a fix for a leak or a prerequisite for a test? Are you sure it's in the right commit?

Reply:
It is needed for test, because otherwise the following goroutine would still persist when the test already finished and then it would get detected by the leak detector.